Assets/Mirror/Core/NetworkIdentity.cs

```diff
@@ -294,10 +294,13 @@ internal static void ResetServerStatics()
         [SerializeField, HideInInspector] bool hasSpawned;
         public bool SpawnedFromInstantiate { get; private set; }
 
-        // NetworkBehaviour components are initialized in Awake once.
-        // Changing them at runtime would get client & server out of sync.
-        // BUT internal so tests can add them after creating the NetworkIdentity
-        internal void InitializeNetworkBehaviours()
+        /// <summary>
+        /// Made public to support gun init. DO NOT USE IN OTHER PLACES, PLEASE.
+        /// NetworkBehaviour components are initialized in Awake once.
+        /// Changing them at runtime would get client & server out of sync.
```

Comment:
Seems a bit sus, but it's an ok workaround. We might consider looking for other solutions in the future... maybe

Reply:
Dunno if we have that much choice here. We can't spawn in fully built guns through Mirror at least, as that would require several hundred different prefabs AFAICT. Leaving it like this for the time being.

Reply:
But it's impossible to call the Setgun method for every client on each player after they are spawned on the client? 
If that doesn't work now I imagine it can work with a refactor maybe, where a networked guncontroller sits on the player on some transform on the player 🤔
I guess we can leave this as is for now, but we might need to look into it later

Reply:
- Calling setgun after spawned on client wouldn't make any difference, as Awake is called on a NetworkIdentity when it is instantiated
- We want to network some modifiers as well, not just the guncontroller